server/src/test/java/org/apache/iotdb/db/tools/IoTDBWatermarkTest.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.tools;
+
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.integration.Constant;
+import org.apache.iotdb.db.service.IoTDB;
+import org.apache.iotdb.db.tools.watermark.WatermarkDetector;
+import org.apache.iotdb.db.utils.EnvironmentUtils;
+import org.apache.iotdb.jdbc.Config;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Notice that, all test begins with "IoTDB" is integration test. All test which will start the
+ * IoTDB server should be defined as integration test.
+ */
+public class IoTDBWatermarkTest {
+
+  private static IoTDB daemon;
+  private static String filePath = "watermarked_query_result.csv";
+  private static PrintWriter writer;
+  private static String secretKey = "ASDFGHJKL";
+  private static String watermarkBitString = "10101000100";
+  private static int mark_rate = 5;
+  private static int max_right_bit = 5;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    EnvironmentUtils.closeStatMonitor();
+    IoTDBDescriptor.getInstance().getConfig().setEnableWatermark(true); // default false
+    IoTDBDescriptor.getInstance().getConfig().setWatermarkSecretKey(secretKey);
+    IoTDBDescriptor.getInstance().getConfig().setWatermarkBitString(watermarkBitString);
+    IoTDBDescriptor.getInstance().getConfig().setWatermarkMethod(String.format("GroupBasedLSBMethod"
+        + "(mark_rate=%d,max_right_bit=%d)", mark_rate, max_right_bit));
+    daemon = IoTDB.getInstance();
+    daemon.active();
+    EnvironmentUtils.envSetUp();
+    Thread.sleep(5000);
```

Comment:
Do not sleep unless it is really necessary.

Reply:
removed